src/main/java/frc/robot/gamepads/XboxControllerMapping.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.gamepads;
+
+import edu.wpi.first.wpilibj.Joystick;
+
+/**
+ * This class maps the inputs for an Xbox style controller.
+ * Tested with:
+ *  - Xbox One Controller (do NOT enable 'Map Gamepad' in simulator)
+ */
+public class XboxControllerMapping implements IControllerMapping {
+
+    public double getLeftX(Joystick joystick) {
+        return joystick.getRawAxis(0);
+    }
+
+    public double getLeftY(Joystick joystick) {
+        return joystick.getRawAxis(1);
+    }
+
+    public double getRightX(Joystick joystick) {
+        return joystick.getRawAxis(2);
+    }
+
+    public double getRightY(Joystick joystick) {
+        return joystick.getRawAxis(3);
+    }
+
+    public boolean getButtonPressed(Joystick joystick, ButtonType buttonType) {
+        switch (buttonType) {
+            case A:
+                return joystick.getRawButton(1);
+            case B:
+                return joystick.getRawButton(2);
+            case X:
+                return joystick.getRawButton(3);
+            case Y:
+                return joystick.getRawButton(4);
+            case LeftBumper:
+                return joystick.getRawButton(5);
+            case LeftTrigger:
+                return joystick.getRawAxis(4) > 0.5;
```

Comment:
Had to make the code a bit more complicated than originally wanted. 
The Xbox controller treats the triggers as axes, but the logitech treats them as buttons.